test-result-summary-client/src/Settings/Settings.jsx

```diff
@@ -168,8 +162,7 @@ export default class Settings extends Component {
                 body: JSON.stringify( { data } ),
             }
 
-            const upsertBuildList = await fetch( `/api/upsertBuildList`, postData );
-            await upsertBuildList.json();
+            await fetchData( `/api/upsertBuildList`);
```

Comment:
The new fetchData function is only for GET request now, but the original code here is for POST request. You can either leave the original code as is, or enhance fetchData to handle more types of requests. Thanks.

Reply:
Thank you pointing this out @LongyuZhang. I am sorry I missed that. What according to you would be a better choice: enhancing fetchData or leaving this particular code piece out as an exception?

Reply:
For this PR, I think you can leave post request code as is, but leave the issue open. 
Then work on enhancing fetchData function to support post request too in another PR.

Reply:
Sure! I have done this and squashed all commits into one. Please let me know if anything else is required. Thank you.